extension/json/json_scan.cpp

```diff
@@ -718,9 +719,21 @@ void JSONScanLocalState::ReadNextBufferSeek(JSONScanGlobalState &gstate, optiona
 		return;
 	}
 
+	auto &raw_handle = file_handle.GetHandle();
+	// For non-on-disk files, we create a handle per thread: this is faster for e.g. S3Filesystem where throttling
+	// per tcp connection can occur meaning that using multiple connections is faster.
+	if (!raw_handle.OnDiskFile() && raw_handle.CanSeek()) {
+		if (!thread_local_filehandle || thread_local_filehandle->GetPath() != raw_handle.GetPath()) {
```

Comment:
I mean It's still the same file right? I don't see how this would be a problem, as we're only using these filehandles to do the range requests. Opening a new file handle to the same file wouldn't really do anything. Either way I'll add some test cases for this!